lib/ansible/plugins/lookup/ipa_vault.py

```diff
@@ -0,0 +1,177 @@
+# (c) 2018, Mario Vazquez <mavazque@redhat.com>
+# (c) 2012-18 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '0.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = """
+    author:
+      - Mario Vazquez (mavazque@redhat.com)
+    lookup: ipa_vault
+    version_added: "2.5"
+    short_description: Gets info from IPA Vault
+    requirements:
+      - ipa client (command line utility)
```

Comment:
This was my first approach, but sadly IPAClient should change the way it works in order to be able to take advantage of it within a lookup plugin.

Basically their post_json method needs a module object to work, in this lookup we don't use this kind of objects. I tried initializing an empty module object and it didn't work because post_json method needs some information present in the object.

On top of that, they should implement cryptographic functions in order to generate the session key needed to interact with the Vault API. 

As a first step we opened an Issue[1] so maybe they can fix the first problem (post_json needing a module object), once this issue is solved we can try to implement this using IPAClient rather than using ipa cli.

[1] https://github.com/ansible/ansible/issues/42616